users/models.py

```diff
@@ -63,7 +64,7 @@ def create_superuser(self, email: str, password: str) -> "CustomUser":
         )
 
 
-class CustomUser(AbstractBaseUser, PermissionsMixin):
+class CustomUser(AbstractBaseUser, PermissionsMixin, BaseModel):
```

Comment:
I'm not sure if new model class shouldn't be as a last to inherit from class because inside of other classes there might be `save` method overridden somewhere. But If in this PR I will find test for this that `full_clean` is called this comment will be invalid

Reply:
Do you think that test that I've added is sufficient to test that?

Reply:
Yea, but please follow things which I wrote about test itself that you should mock `full_clean` method instead of whole model`